src/WindowsUtils/BusyLoopLauncherTest.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/base/casts.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <atomic>
+
+#include "OrbitBase/ExecutablePath.h"
+#include "OrbitBase/GetLastError.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "Test/Path.h"
+#include "TestUtils/TestUtils.h"
+#include "WindowsUtils/BusyLoopLauncher.h"
+#include "WindowsUtils/BusyLoopUtils.h"
+#include "WindowsUtils/DllInjection.h"
+#include "WindowsUtils/OpenProcess.h"
+#include "WindowsUtils/ProcessList.h"
+
+namespace orbit_windows_utils {
+
+namespace {
+
+using orbit_test_utils::HasError;
+using orbit_test_utils::HasNoError;
+using orbit_windows_utils::BusyLoopInfo;
+using orbit_windows_utils::BusyLoopLauncher;
+using orbit_windows_utils::OpenProcess;
+using orbit_windows_utils::ProcessList;
+using orbit_windows_utils::SafeHandle;
+
+[[nodiscard]] std::filesystem::path GetTestExecutablePath() {
+  static auto path = orbit_base::GetExecutableDir() / "FakeCliProgram.exe";
+  return path;
+}
+
+[[nodiscard]] std::filesystem::path GetTestDllPath() {
+  return orbit_test::GetTestdataDir() / "libtest.dll";
+}
+
+__declspec(noinline) void IncrementCounter(uint32_t* counter, std::atomic<bool>* exit_requested) {
+  ++(*counter);
+
+  // Wait for exit request.
+  while (!(*exit_requested)) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+  }
+}
+
+}  // namespace
+
+TEST(BusyLoop, BusyLoopLauncher) {
+  // Start process that would normally exit instantly and install busy loop at entry point.
+  BusyLoopLauncher busy_loop_launcher;
+  ErrorMessageOr<BusyLoopInfo> result = busy_loop_launcher.StartWithBusyLoopAtEntryPoint(
+      GetTestExecutablePath(), /*working_directory=*/"", /*arguments=*/"");
+  ASSERT_TRUE(result.has_value());
+
+  // Validate returned busy loop info.
+  const BusyLoopInfo& busy_loop_info = result.value();
+  ASSERT_TRUE(busy_loop_info.original_bytes.size() > 0);
+  ASSERT_NE(busy_loop_info.process_id, 0);
+  ASSERT_NE(busy_loop_info.address, 0);
+
+  // At this point, the process should be busy looping at entry point.
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Is this sleep needed for anything? Otherwise I'd remove it - always looks like a code smell.

Reply:
Removed the sleep. I wanted to make sure the process was actually busy looping, but it's redundant as other steps depend on the process being still alive below.